### ISSUE TITLE
Support pip's editable package 'subdirectory' component

### ISSUE THREAD

**Issue by chris-griffin** [snyk/snyk#109]
- `node -v`:
`v8.1.1`

- `npm -v`:
`5.0.3`

- `snyk -v`:
`1.43.0`

- Command run:
    `snyk test --file=common.txt --packageManager=pip`

    where common.txt contains a [`subdirectory`](https://pip.pypa.io/en/stable/reference/pip_install/#vcs-support) component like below

    ```py
    -e git+https://aksdfidfed866d2a4470d1e077f5f@github.com/example/example.git@asdflkjasdfkj8ebc0a93200879c988b43a96#egg=example&subdirectory=example
    ```

    whereas the same example without a `subdirectory` component will work
    ```py
     -e git+https://aksdfidfed866d2a4470asdfasdf@github.com/example/example.git@asdflkjasdfkj8ebc0a932asdfasdf3a96#egg=example
    ```

### Expected behaviour
To support the subdirectory component

### Actual behaviour
Errors:

```sh
Traceback (most recent call last):
  File "/usr/local/lib/node_modules/snyk/node_modules/snyk-python-plugin/plug/pip_resolve.py", line 117, in <module>
    sys.exit(create_dependencies_tree_by_req_file_path(sys.argv[1]))
  File "/usr/local/lib/node_modules/snyk/node_modules/snyk-python-plugin/plug/pip_resolve.py", line 104, in create_dependencies_tree_by_req_file_path
    required = get_requirements_list(requirements_file)
  File "/usr/local/lib/node_modules/snyk/node_modules/snyk-python-plugin/plug/pip_resolve.py", line 86, in get_requirements_list
    req_list = list(requirements.parse(requirements_file))
  File "/usr/local/lib/node_modules/snyk/node_modules/snyk-python-plugin/plug/requirements/parser.py", line 53, in parse
    yield Requirement.parse(line)
  File "/usr/local/lib/node_modules/snyk/node_modules/snyk-python-plugin/plug/requirements/requirement.py", line 179, in parse
    re.sub(r'^(-e|--editable=?)\s*', '', line))
  File "/usr/local/lib/node_modules/snyk/node_modules/snyk-python-plugin/plug/requirements/requirement.py", line 108, in parse_editable
    assert local_match is not None, 'This should match everything'
AssertionError: This should match everything
```

### Steps to reproduce

1. `npm install -g snyk`
2. `snyk auth`
3. Create common.txt with the following contents
    ```py
    -e git+https://aksdfidfed866d2a4470d1e077f5f@github.com/example/example.git@asdflkjasdfkj8ebc0a93200879c988b43a96#egg=example&subdirectory=example
    ```
4. `snyk test --file=common.txt --packageManager=pip`

---

If applicable, please append the `--debug` flag on your command and include the output here **ensuring to remove any sensitive/personal details or tokens.

```sh
snyk test --file=common.txt --packageManager=pip --debug
  snyk test { _: [ [Circular] ],
  file: 'common.txt',
  packageManager: 'pip',
  debug: true } +0ms
  snyk checking for cli updates +274ms
  snyk add local true +51ms
  snyk add packageManager pip +0ms
  snyk add error-message Traceback (most recent call last):
  File "/usr/local/lib/node_modules/snyk/node_modules/snyk-python-plugin/plug/pip_resolve.py", line 117, in <module>
    sys.exit(create_dependencies_tree_by_req_file_path(sys.argv[1]))
  File "/usr/local/lib/node_modules/snyk/node_modules/snyk-python-plugin/plug/pip_resolve.py", line 104, in create_dependencies_tree_by_req_file_path
    required = get_requirements_list(requirements_file)
  File "/usr/local/lib/node_modules/snyk/node_modules/snyk-python-plugin/plug/pip_resolve.py", line 86, in get_requirements_list
    req_list = list(requirements.parse(requirements_file))
  File "/usr/local/lib/node_modules/snyk/node_modules/snyk-python-plugin/plug/requirements/parser.py", line 41, in parse
    for requirement in parse(f):
  File "/usr/local/lib/node_modules/snyk/node_modules/snyk-python-plugin/plug/requirements/parser.py", line 53, in parse
    yield Requirement.parse(line)
  File "/usr/local/lib/node_modules/snyk/node_modules/snyk-python-plugin/plug/requirements/requirement.py", line 179, in parse
    re.sub(r'^(-e|--editable=?)\s*', '', line))
  File "/usr/local/lib/node_modules/snyk/node_modules/snyk-python-plugin/plug/requirements/requirement.py", line 108, in parse_editable
    assert local_match is not None, 'This should match everything'
AssertionError: This should match everything
 +769ms
  snyk add error Error: Traceback (most recent call last):
  File "/usr/local/lib/node_modules/snyk/node_modules/snyk-python-plugin/plug/pip_resolve.py", line 117, in <module>
    sys.exit(create_dependencies_tree_by_req_file_path(sys.argv[1]))
  File "/usr/local/lib/node_modules/snyk/node_modules/snyk-python-plugin/plug/pip_resolve.py", line 104, in create_dependencies_tree_by_req_file_path
    required = get_requirements_list(requirements_file)
  File "/usr/local/lib/node_modules/snyk/node_modules/snyk-python-plugin/plug/pip_resolve.py", line 86, in get_requirements_list
    req_list = list(requirements.parse(requirements_file))
  File "/usr/local/lib/node_modules/snyk/node_modules/snyk-python-plugin/plug/requirements/parser.py", line 41, in parse
    for requirement in parse(f):
  File "/usr/local/lib/node_modules/snyk/node_modules/snyk-python-plugin/plug/requirements/parser.py", line 53, in parse
    yield Requirement.parse(line)
  File "/usr/local/lib/node_modules/snyk/node_modules/snyk-python-plugin/plug/requirements/requirement.py", line 179, in parse
    re.sub(r'^(-e|--editable=?)\s*', '', line))
  File "/usr/local/lib/node_modules/snyk/node_modules/snyk-python-plugin/plug/requirements/requirement.py", line 108, in parse_editable
    assert local_match is not None, 'This should match everything'
AssertionError: This should match everything

    at /usr/local/lib/node_modules/snyk/node_modules/snyk-python-plugin/lib/index.js:48:13
    at <anonymous>
    at process._tickCallback (internal/process/next_tick.js:169:7) +0ms
  snyk add error-code undefined +0ms
  snyk add command test +0ms
Error: Traceback (most recent call last):
  File "/usr/local/lib/node_modules/snyk/node_modules/snyk-python-plugin/plug/pip_resolve.py", line 117, in <module>
    sys.exit(create_dependencies_tree_by_req_file_path(sys.argv[1]))
  File "/usr/local/lib/node_modules/snyk/node_modules/snyk-python-plugin/plug/pip_resolve.py", line 104, in create_dependencies_tree_by_req_file_path
    required = get_requirements_list(requirements_file)
  File "/usr/local/lib/node_modules/snyk/node_modules/snyk-python-plugin/plug/pip_resolve.py", line 86, in get_requirements_list
    req_list = list(requirements.parse(requirements_file))
  File "/usr/local/lib/node_modules/snyk/node_modules/snyk-python-plugin/plug/requirements/parser.py", line 41, in parse
    for requirement in parse(f):
  File "/usr/local/lib/node_modules/snyk/node_modules/snyk-python-plugin/plug/requirements/parser.py", line 53, in parse
    yield Requirement.parse(line)
  File "/usr/local/lib/node_modules/snyk/node_modules/snyk-python-plugin/plug/requirements/requirement.py", line 179, in parse
    re.sub(r'^(-e|--editable=?)\s*', '', line))
  File "/usr/local/lib/node_modules/snyk/node_modules/snyk-python-plugin/plug/requirements/requirement.py", line 108, in parse_editable
    assert local_match is not None, 'This should match everything'
AssertionError: This should match everything

    at /usr/local/lib/node_modules/snyk/node_modules/snyk-python-plugin/lib/index.js:48:13
    at <anonymous>
    at process._tickCallback (internal/process/next_tick.js:169:7)
  snyk { command: 'bad-command',
  snyk   args: [ { file: 'common.txt', packageManager: 'pip', debug: true } ],
  snyk   metadata: 
  snyk    { local: true,
  snyk      packageManager: 'pip',
  snyk      'error-message': 'Traceback (most recent call last):\n  File "/usr/local/lib/node_modules/snyk/node_modules/snyk-python-plugin/plug/pip_resolve.py", line 117, in <module>\n    sys.exit(create_dependencies_tree_by_req_file_path(sys.argv[1]))\n  File "/usr/local/lib/node_modules/snyk/node_modules/snyk-python-plugin/plug/pip_resolve.py", line 104, in create_dependencies_tree_by_req_file_path\n    required = get_requirements_list(requirements_file)\n  File "/usr/local/lib/node_modules/snyk/node_modules/snyk-python-plugin/plug/pip_resolve.py", line 86, in get_requirements_list\n    req_list = list(requirements.parse(requirements_file))\n  File "/usr/local/lib/node_modules/snyk/node_modules/snyk-python-plugin/plug/requirements/parser.py", line 41, in parse\n    for requirement in parse(f):\n  File "/usr/local/lib/node_modules/snyk/node_modules/snyk-python-plugin/plug/requirements/parser.py", line 53, in parse\n    yield Requirement.parse(line)\n  File "/usr/local/lib/node_modules/snyk/node_modules/snyk-python-plugin/plug/requirements/requirement.py", line 179, in parse\n    re.sub(r\'^(-e|--editable=?)\\s*\', \'\', line))\n  File "/usr/local/lib/node_modules/snyk/node_modules/snyk-python-plugin/plug/requirements/requirement.py", line 108, in parse_editable\n    assert local_match is not None, \'This should match everything\'\nAssertionError: This should match everything\n',
  snyk      error: 'Error: Traceback (most recent call last):\n  File "/usr/local/lib/node_modules/snyk/node_modules/snyk-python-plugin/plug/pip_resolve.py", line 117, in <module>\n    sys.exit(create_dependencies_tree_by_req_file_path(sys.argv[1]))\n  File "/usr/local/lib/node_modules/snyk/node_modules/snyk-python-plugin/plug/pip_resolve.py", line 104, in create_dependencies_tree_by_req_file_path\n    required = get_requirements_list(requirements_file)\n  File "/usr/local/lib/node_modules/snyk/node_modules/snyk-python-plugin/plug/pip_resolve.py", line 86, in get_requirements_list\n    req_list = list(requirements.parse(requirements_file))\n  File "/usr/local/lib/node_modules/snyk/node_modules/snyk-python-plugin/plug/requirements/parser.py", line 41, in parse\n    for requirement in parse(f):\n  File "/usr/local/lib/node_modules/snyk/node_modules/snyk-python-plugin/plug/requirements/parser.py", line 53, in parse\n    yield Requirement.parse(line)\n  File "/usr/local/lib/node_modules/snyk/node_modules/snyk-python-plugin/plug/requirements/requirement.py", line 179, in parse\n    re.sub(r\'^(-e|--editable=?)\\s*\', \'\', line))\n  File "/usr/local/lib/node_modules/snyk/node_modules/snyk-python-plugin/plug/requirements/requirement.py", line 108, in parse_editable\n    assert local_match is not None, \'This should match everything\'\nAssertionError: This should match everything\n\n    at /usr/local/lib/node_modules/snyk/node_modules/snyk-python-plugin/lib/index.js:48:13\n    at <anonymous>\n    at process._tickCallback (internal/process/next_tick.js:169:7)',
  snyk      'error-code': undefined,
  snyk      command: 'test' },
  snyk   version: '1.43.0',
  snyk   os: 'OS X',
  snyk   nodeVersion: 'v8.1.1',
  snyk   id: 'redacted',
  snyk   ci: false } +3ms
```

**Comment by darscan** [comment]
@chris-griffin There's a new release going out shortly. There was a minor hold up due to some unrelated fixes that needed to land first.

**Comment by darscan** [comment]
Fixed in `v1.44.1`. Thanks for reporting and helping to fix.
